main.go

```diff
@@ -14,31 +14,32 @@ var (
 	input_text = flag.String("input-text", "Default input text", "Input text to print.")
 )
 
-type Result = struct {
-	Pipeline    *beam.Pipeline
-	Scope       beam.Scope
-	PCollection beam.PCollection
+func init() {
+	// DoFns should be registered with Beam.
+	beam.RegisterFunction(printAndEmit)
 }
 
-func my_pipeline(input_text string) Result {
-	beam.Init()
-
-	pipeline, scope := beam.NewPipelineWithRoot()
+func printAndEmit(element string, emit func(string)) {
+	fmt.Println(element)
+	emit(element)
+}
 
+func my_pipeline(scope beam.Scope, input_text string) beam.PCollection {
 	elements := beam.Create(scope, "Hello", "World!", input_text)
-	elements = beam.ParDo(scope, func(elem string, emit func(string)) { fmt.Println(elem); emit(elem) }, elements)
-
-	return Result{pipeline, scope, elements}
+	elements = beam.ParDo(scope, printAndEmit, elements)
+	return elements
```

Comment:
Style wise, this is a code smell. Simpler to simply return without the variable.
```suggestion
	return beam.ParDo(scope, printAndEmit, elements)
```